tidy3d/components/monitor.py

```diff
@@ -270,6 +271,46 @@ def _storage_size_solver(self, num_cells: int, tmesh: ArrayFloat1D) -> int:
         return solver_data_size
 
 
+class AbstractAuxFieldMonitor(Monitor, ABC):
+    """:class:`Monitor` that records electromagnetic field data as a function of x,y,z."""
+
+    fields: Tuple[AuxField, ...] = pydantic.Field(
+        (),
+        title="Aux Field Components",
+        description="Collection of auxiliary field components to store in the monitor.",
+    )
```

Comment:
How can the user know what's allowed to enter here?

I mean, I kind of understand that the check can only happen on the Simulation level. So my question is maybe mostly documentation wise. If I'm trying to understand what to put here, I need to discover two things: the possible values and what they mean, for which I need to look at the AuxFieldTimeMonitorData, and what is allowed in my simulation, for which I need to look at the `Medium.aux_fields` in the mediums I'm using.

How do we assemble all that in coherent instructions to the user?

Reply:
Hmm, yeah. Currently, when building the monitor, you can at least see the list of all possible aux fields (including those used in nonlinearities not present in the simulation). Then when building the simulation, it gives a warning if some of these aux fields are not actually present in the simulation -- in which case the monitor just records them as zero.

I couldn't really think of anything cleaner -- the Monitor can't really know about the Simulation. I can make `Simulation.aux_fields` public (it lists all the aux fields actually present in the simulation, compiled from all the Medums).

Reply:
I guess I can also improve this docstring to state this explicitly

Reply:
Yeah maybe explicitly reference the data components where the various docstrings for the aux fields live?